src/mbgl/map/transform_state.cpp

```diff
@@ -104,18 +161,59 @@ void TransformState::getProjMatrix(mat4& projMatrix, uint16_t nearZ, bool aligne
     }
 }
 
+void TransformState::updateMatricesIfNeeded() const {
+    if (!needMatricesUpdate() || size.isEmpty()) return;
+
+    getProjMatrix(projectionMatrix);
+    coordMatrix = coordinatePointMatrix(getProjectionMatrix());
```

Comment:
here it's better to access `projectionMatrix` and etc directly, otherwise `updateMatricesIfNeeded()` will be called recurrently..

Reply:
Right! Thanks!